reflect.go

```diff
@@ -106,6 +131,147 @@ func inspect(t reflect.Type, jsonTag string) Property {
+	//  or mapType.Elem().Kind() ==> *Struct|Struct|*string|string
+	if mapType.Elem().Kind().String() != "interface" {
+		isSlice := isSliceOrArryType(mapType.Elem().Kind())
+		if isSlice || isByteArrayType(mapType.Elem()) {
+			// if map value is slice
+			// Example:
+			//   mapType.Elem()==> []*Struct|Struct|*string|string
+			mapType = mapType.Elem()
+		}
+
+		// if map value is struct or built-in primitive type
+		// Example:
+		//    mapType.Elem()==> *Struct|Struct|*string|string
+		isPrimitive := isPrimitiveType(mapType.Elem().Name(), mapType.Elem())
+
+		if isByteArrayType(mapType.Elem()) {
+			prop.Type = "string"
+		} else {
+			if isSlice {
+				prop.Type = types.Array.String()
+				prop.Items = &Items{}
+				if isPrimitive {
+					prop.Items.Type = jsonSchemaType(mapType.Elem().String(), mapType.Elem())
+					prop.GoType = getGoType(mapType.Elem())
+				} else {
+					prop.Items.Ref = makeMapRef(mapType.Elem().String())
+					prop.GoType = getGoType(mapType.Elem())
+				}
+			} else if isPrimitive {
+				prop.Type = jsonSchemaType(mapType.Elem().String(), mapType.Elem())
+				prop.GoType = getGoType(mapType.Elem())
+			} else {
+				prop.Ref = makeMapRef(mapType.Elem().String())
+				prop.GoType = getGoType(mapType.Elem())
+			}
+		}
+	}
+
+	return &prop
+}
+
+func getGoType(t reflect.Type) reflect.Type {
+
+	var goType reflect.Type
+
+	if t.Kind() == reflect.Ptr {
+		goType = t.Elem()
+	} else {
+		goType = t
+	}
+
+	return goType
+}
+
+func makeMapRef(typeName string) string {
+	type_name := strings.Trim(typeName, "*")
+	return makeRef(type_name)
+}
+
+func isSliceOrArryType(t reflect.Kind) bool {
+	return t == reflect.Slice || t == reflect.Array
+}
+
+func isByteArrayType(t reflect.Type) bool {
+	return (t.Kind() == reflect.Slice || t.Kind() == reflect.Array) &&
+		t.Elem().Kind() == reflect.Uint8
+}
+
+// isPrimitiveType Whether it is a built-in primitive type
+func isPrimitiveType(modelName string, modelType reflect.Type) bool {
+	var modelKind reflect.Kind
+
+	if modelType.Kind() == reflect.Ptr {
+		modelKind = modelType.Elem().Kind()
+	} else {
+		modelKind = modelType.Kind()
+	}
+
+	switch modelKind {
+	case reflect.Bool:
+		return true
+	case reflect.Float32, reflect.Float64,
+		reflect.Int, reflect.Int8, reflect.Int16, reflect.Int32, reflect.Int64,
+		reflect.Uint, reflect.Uint8, reflect.Uint16, reflect.Uint32, reflect.Uint64:
+		return true
+	case reflect.String:
+		return true
+	}
+
+	if len(modelName) == 0 {
+		return false
+	}
+
+	return strings.Contains("time.Time time.Duration json.Number", modelName)
+}
+
+func jsonSchemaType(modelName string, modelType reflect.Type) string {
+	var modelKind reflect.Kind
+
+	if modelType.Kind() == reflect.Ptr {
+		modelKind = modelType.Elem().Kind()
+	} else {
+		modelKind = modelType.Kind()
+	}
+
+	schemaMap := map[string]string{
+		"time.Time":     "string",
+		"time.Duration": "integer",
+		"json.Number":   "number",
+	}
```

Comment:
Put it in a global variable and only need to initialize it once.

Reply:
ok resolved it